src/ring_theory/integral_closure.lean

```diff
@@ -281,3 +282,87 @@ begin
   replace hmp := congr_arg subtype.val hmp,
   exact subtype.eq hmp
 end
+
+end
+
+section algebra
+open algebra
+variables {R : Type*} {A : Type*} {B : Type*}
+variables [comm_ring R] [comm_ring A] [comm_ring B]
+variables [algebra R A] [algebra A B]
+variables (A_alg : ∀ x : A, is_integral R x)
+
+include A_alg
+
+set_option class.instance_max_depth 50
+
+lemma is_integral_trans_aux (x : B) {p : polynomial A} (pmonic : monic p) (hp : aeval A B x p = 0)
+  (S : set (comap R A B))
+  (hS : S = (↑((finset.range (p.nat_degree + 1)).image
+    (λ i, to_comap R A B (p.coeff i))) : set (comap R A B))) :
```

Comment:
Presumably this is okay but I'm not following. `p : polynomial A` so `p.coeff i : A` but doesn't `to_comap R A B` want a `B`? Is there a coercion from `A` to `B` from the algebra?

Reply:
The type signature of `to_comap` is: `S →ₐ[R] comap R S A`.
Does that clear things up?

Reply:
Ah, there are two of them!
```lean
def comap.to_comap : A → comap R S A := id
def to_comap : S →ₐ[R] comap R S A := [...]
```
Maybe these names aren't the best?

Reply:
Maybe not... but this PR is already touching lots of things... I'd rather park this problem.